linera-views/tests/key_value_store_view.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use linera_views::{
+    key_value_store_view::KeyValueStoreView,
+    memory::create_memory_context,
+    views::{CryptoHashRootView, RootView, View},
+};
+use rand::{distributions::Uniform, Rng, SeedableRng};
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fmt::Debug,
+};
+
+#[derive(CryptoHashRootView)]
+pub struct StateView<C> {
+    pub store: KeyValueStoreView<C>,
+}
+
+fn remove_by_prefix<V: Debug>(map: &mut BTreeMap<Vec<u8>, V>, key_prefix: Vec<u8>) {
+    map.retain(|key, _| !key.starts_with(&key_prefix));
+}
+
+fn total_size(vec: &Vec<(Vec<u8>, Vec<u8>)>) -> (u64, u64) {
+    let mut total_key_size = 0;
+    let mut total_value_size = 0;
+    for (key, value) in vec {
+        total_key_size += key.len();
+        total_value_size += value.len();
+    }
+    (total_key_size as u64, total_value_size as u64)
+}
+
+#[tokio::test]
+async fn key_value_store_view_mutability() {
+    let context = create_memory_context();
+    let mut rng = rand::rngs::StdRng::seed_from_u64(2);
+    let mut state_map = BTreeMap::new();
+    let n = 200;
+    let mut all_keys = BTreeSet::new();
+    for _ in 0..n {
+        let mut view = StateView::load(context.clone()).await.unwrap();
+        let save = rng.gen::<bool>();
+        let read_state = view.store.index_values().await.unwrap();
+        let state_vec = state_map.clone().into_iter().collect::<Vec<_>>();
+        assert_eq!(state_vec, read_state);
+        assert_eq!(total_size(&state_vec), view.store.total_size());
+        //
+        let count_oper = rng.gen_range(0..25);
+        let mut new_state_map = state_map.clone();
+        let mut new_state_vec = state_vec.clone();
+        for _ in 0..count_oper {
+            let choice = rng.gen_range(0..5);
+            let count = view.store.count().await.unwrap();
+            if choice == 0 {
+                // inserting random stuff
+                let n_ins = rng.gen_range(0..10);
+                for _ in 0..n_ins {
+                    let len = rng.gen_range(1..6);
+                    let key = rng
+                        .clone()
```

Comment:
```suggestion
                    let key = (&mut rng)
```

(Otherwise we fail to modify the state of the `rng`, and it would yield the same result if we did this again.)

Reply:
Ok, thanks.